operator/src/main/resources/scripts/modelInImage.sh

```diff
@@ -849,6 +932,109 @@ function wdtUpdateModelDomain() {
   trace "Exiting wdtUpdateModelDomain"
 }
 
+function wdtHandleOnlineUpdate() {
+
+  trace "Entering wdtHandleOnlineUpdate"
+  # wdt shell script may return non-zero code if trap is on, then it will go to trap instead
+  # temporarily disable it
+  stop_trap
+  if [ -z ${MII_USE_ONLINE_UPDATE} ] || [ "false" == "${MII_USE_ONLINE_UPDATE}" ] ; then
+    # no op for offline use case
+    return
+  fi
+
+  # We need to extract all the archives, WDT online checks for file existence
+  # even for delete
+  #
+  mkdir -p ${DOMAIN_HOME}/lib || exitOrLoop
+  for file in $(sort_files ${IMG_ARCHIVES_ROOTDIR} "*.zip")
+    do
+        # expand the archive domain libraries to the domain lib
+        cd ${DOMAIN_HOME}/lib || return exitOrLoop
+        ${JAVA_HOME}/bin/jar xf ${IMG_ARCHIVES_ROOTDIR}/${file} wlsdeploy/domainLibraries/
+
+        if [ $? -ne 0 ] ; then
+          trace SEVERE  "Domain Source Type is FromModel, error in extracting domain libs ${IMG_ARCHIVES_ROOTDIR}/${file}"
+          exitOrLoop
+        fi
+
+        # expand the archive apps and shared lib to the wlsdeploy/* directories
+        # the config.xml is referencing them from that path
+
+        cd ${DOMAIN_HOME} || exitOrLoop
+        ${JAVA_HOME}/bin/jar xf ${IMG_ARCHIVES_ROOTDIR}/${file} wlsdeploy/
+
+        if [ $? -ne 0 ] ; then
+          trace SEVERE "Domain Source Type is FromModel, error in extracting application archive ${IMG_ARCHIVES_ROOTDIR}/${file}"
+          exitOrLoop
+        fi
+
+
+    done
+
+
+  # Save off the encrypted model
+  cp ${DOMAIN_HOME}/wlsdeploy/domain_model.json /tmp/encrypted_merge_model.json
+  local admin_user=$(cat /weblogic-operator/secrets/username)
+  local admin_pwd=$(cat /weblogic-operator/secrets/password)
+
+  if [ -z ${AS_SERVICE_NAME} ] || [ -z ${ADMIN_PORT} ] ; then
+    trace SEVERE "Cannot find admin service name or port"
+    exitOrLoop
+  fi
+
+  local admin_url
+  if [ -z "${ADMIN_PORT_SECURE}" ] ; then
+    admin_url="t3://${AS_SERVICE_NAME}:${ADMIN_PORT}"
+  else
+    admin_url="t3s://${AS_SERVICE_NAME}:${ADMIN_PORT}"
+  fi
+  echo ${admin_pwd} | ${WDT_BINDIR}/updateDomain.sh -oracle_home ${MW_HOME} \
+   -admin_url ${admin_url} -admin_user ${admin_user} -model_file \
+   /tmp/diffed_model.yaml -domain_home ${DOMAIN_HOME} ${archive_list} \
+   -discard_current_edit -output_dir /tmp  >  ${WDT_OUTPUT} 2>&1
+
+  local ret=$?
+
+  trace "Completed online update="${ret}
+  if [ ${ret} -eq ${PROG_RESTART_REQUIRED} ] ; then
+    write_updatedresult ${ret}
+    write_non_dynamic_changes_text_file
+  elif [ ${ret} -ne 0 ] ; then
+    trace SEVERE "Online update failed. Check error in the logs " \
+       "Note: Changes in the optional configmap and/or image may needs to be corrected"
```

Comment:
Needs a little wordsmithing. Let's meet to discuss. Also, how can we get to this point if the image changed? Image changes always result in an offline/roll...

Reply:
Done.